### PR TITLE
Use Community Types for agGrid

### PR DESCRIPTION
**Pull request by lbwexler**
I am limiting this to typescript related changes in order to faciliate merge/release before worrying about some of the warts we discovered with our use of `renderer` and `getValueFn`.

We can deal with that in a seperate PR or release.  Will create a ticket with findings related to that now.

Note there is one TODO, related to a change with behavior of sort state for pivot grids.  this still needs to be tested/addressed.

Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [X] Caught up with `develop` branch as of last change.
- [X] Added CHANGELOG entry, or determined not required.
- [X] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [X] Updated doc comments / prop-types, or determined not required.
- [X] Reviewed and tested on Mobile, or determined not required.
- [X] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: cmp/ag-grid/AgGridModel.ts [hunk 1]
```diff
@@ -347,7 +349,8 @@ export class AgGridModel extends HoistModel {
 
             // finally apply sorts from state to secondary columns
             secondaryColumnState.forEach(state => {
-                const col = colApi.getSecondaryPivotColumn(state.colId[0], state.colId[1]);
+                // TODO: Check this -- was very off
```
Comment (ghsolomon) [inline cmp/ag-grid/AgGridModel.ts:352]: Is this TODO still relevant?

File: cmp/grid/Grid.ts [hunk 2]
```diff
@@ -201,10 +213,10 @@ class GridLocalModel extends HoistModel {
                 clipboardCopy: Icon.copy({asHtml: true})
             },
             components: {
-                agColumnHeader: props => columnHeader({gridModel: model, ...props}),
-                agColumnGroupHeader: props => columnGroupHeader(props)
+                agColumnHeader: columnHeader,
```
Comment (ghsolomon) [inline cmp/grid/Grid.ts:216]: Is `gridModel` not a required prop of `ColumnHeader`? I see it referenced in `ColumnHeaderModel.gridModel`

File: cmp/grid/Grid.ts [hunk 3]
```diff
@@ -201,10 +213,10 @@ class GridLocalModel extends HoistModel {
...
+                agColumnGroupHeader: columnGroupHeader
             },
-            rowSelection: selModel.mode,
+            rowSelection: selModel.mode == 'disabled' ? undefined : selModel.mode,
```
Comment (ghsolomon) [inline cmp/grid/Grid.ts:219]: Makes sense-  Do we need to advertise this as a 'breaking change'?

File: cmp/grid/Grid.ts (hunk 2 above)
Comment (lbwexler) [reply cmp/grid/Grid.ts:216]: thanks for the careful look -- actually getting installed by agGrid through a special property on column:

 headerComponentParams: {gridModel, xhColumn: this},

File: cmp/grid/Grid.ts (hunk 3 above)
Comment (lbwexler) [reply cmp/grid/Grid.ts:219]: was not intended to be a breaking change!  Will test this more carefully -- my assumption was that this was just a change to satisfy typescript

File: cmp/grid/Grid.ts [hunk 4]
```diff
@@ -587,7 +599,9 @@ class GridLocalModel extends HoistModel {
             // Refresh cells in columns with complex renderers
             const refreshCols = visibleCols.filter(c => c.rendererIsComplex);
             if (!isEmpty(refreshCols)) {
-                const rowNodes = compact(transaction.update.map(r => agApi.getRowNode(r.agId))),
+                const rowNodes = transaction.update
+                        .map(r => agApi.getRowNode(r.agId))
+                        .filter(n => n != null),
```
[the remark this answers is not in the document]
Comment (lbwexler) [reply cmp/grid/Grid.ts:604]: typescript was not following it.

File: cmp/ag-grid/AgGridModel.ts (hunk 1 above)
Comment (lbwexler) [reply cmp/ag-grid/AgGridModel.ts:352]: yes -- just changed this code without running it -- need to test pivot grids still

File: cmp/grid/impl/ColumnHeader.ts [hunk 5]
```diff
@@ -146,15 +154,19 @@ class ColumnHeaderModel extends HoistModel {
...
+    get xhColumn(): Column {
         return this.componentProps.xhColumn;
     }
+
     get agColumn() {
         return this.componentProps.column;
     }
+
     get colId() {
         return this.agColumn.colId;
```
Comment (ghsolomon) [inline cmp/grid/impl/ColumnHeader.ts:167]: Should this be `getColId()`? Just checking since I saw it was changed in several other places.

Comment (lbwexler) [reply cmp/grid/impl/ColumnHeader.ts]: fixed -- thanks

File: cmp/ag-grid/AgGridModel.ts (hunk 1 above)
Comment (lbwexler) [reply cmp/ag-grid/AgGridModel.ts]: restored the code -- this was already broken, and not looking to change runtime behavior in this change.  Will file a ticket

**Pull request merged**